framework/src/org/checkerframework/framework/source/SourceChecker.java

```diff
@@ -427,6 +428,15 @@ public final synchronized void init(ProcessingEnvironment env) {
         // The processingEnvironment field will also be set by the superclass' init method.
         // This is used to trigger AggregateChecker's setProcessingEnvironment.
         setProcessingEnvironment(env);
+
+        double jreVersion = PluginUtil.getJreVersion();
+        if (jreVersion != 1.8) {
+            report(
+                    org.checkerframework.framework.source.Result.failure(
+                            "The Checker Framework must be run under JDK 1.8.  You are using version %f.",
+                            jreVersion),
+                    null);
```

Comment:
This uses `null` as the location at which to issue the warning.
Is there a better way to issue the warning, such as by immediately terminating compilation?

Reply:
Yes, use ErrorReporter.errorAbort().  That method throws an exception.